rep-0146.rst

```diff
@@ -0,0 +1,80 @@
+REP: 146
+Title: ROS Robot Parameter Naming Convention
+Author: Kei Okada <k-okada@jsk.t.u-tokyo.ac.jp>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 31-Jly-2015
+Post-History:
+
+Abstract
+========
+
+This REP aims to define the best practices for robot related parameter
+name space for making robot-agnostic packages.
+
+
+Motivation
+==========
+
+As the number of ROS-supported robot increases, we will write
+robot-agnostic packages, but still they need robot-specific
+information as calibration parameters.
+
+A `robot_description` parameter is widely shared naming convention
+among ROS community and this REP proposes such conventions for robot specific information on
+parameter, so application program can build their program on top of that.
+
+
+Specifications
+==============
+
+Parameters
+----------
+
+robot_description
+'''''''''''''''''
+
+`robot_description` stores robot kinematics / geometry information using
+URDF format.
+
+
+robot_type
+''''''''''
+
+`robot_type` stores robot type in lower cases, for example 'pr2' in PR2 robot case [1], 'pepper' in Pepper robot case and 'fetch' in Fetch robot cases. This could be regarded as manufacturer's model name.
```

Comment:
Can you add a justification for lower case? It seems that `PR2` would be fine? I could also see manufactures considering adding some level of versioning in here. For example 

http://library.isr.ist.utl.pt/docs/roswiki/Robots(2f)Kingfisher.html vs https://robots.ros.org/kingfisher/ would likely want to be distinguished as something like `Kingfisher M100` vs `Kingfisher` (though it does appear to be renamed the Heron for disambiguation now)

Reply:
I'd suggest adding something like 
```
The `robot_type` should capture the name of the class of robot in a way that a user would understand the scope of the robot. If it's a commercial product it is recommended to use the production model name. ` Then move onto the examples. 
```

Reply:
I  just to avoid confusion, for example people prefer the code like  `rospy.get_param("robot/name") == "pr2"`  than   `rospy.get_param("robot/name") == "PR2"`.
or, if wen can use `$(param)` tag in future, some one want to change file name from `PR2_configuration.yaml` to `pr2_configuration.yaml`. So if we decide we only allow lower case, we can get rid of these kind of discussion.
But I found that negative cases, which is "AIBO" (old one) and "aibo" (recent ROS version)..... so  I removed the sentence.

> I could also see manufactures considering adding some level of versioning in here.

That's difficult question. Should we use "AIBO" or "AIBO ERS-110"/"AIBO ERS-21"?
In our cases "HRP-2" or "HRP2JSK" (which is student modified version of original HRP-2 robot)...

But I think important thing is that we can programmably get the name of the current robot and run 'case' function to choose appropriate function.